### ISSUE TITLE
Results missing for a few areas

### ISSUE THREAD

**Issue by rafapereirabr** [ipeaGIT/accessibility#40]
Working on this accessibility analysis provided by @ECDelmelle , we've found what seems to be a strange bug. The analysis is documented in the reproducible example below.

# Problem
The output of `accessibility::cumulative_cutoff()` returns 46 areas with missing values. However, the results look totally fine when we calculate accessibility directly with `r5r::accessibility()`.

# Reprex

## Downloading the data and building the transport network
```{r}
options(java.parameters = '-Xmx18G')
library(r5r)
library(lehdr)
library(tidycensus)
library(sf)
library(tigris)
library(tidyverse)
library(accessibility)
library(httr)

# Get the number of workers data from the Census and filter out all jobs, low-skill and high-skill jobs
pa_wac <- grab_lodes(state = "pa",
                     year = 2020,
                     lodes_type = "wac",
                     job_type = "JT01",
                     segment = 'S000',
                     agg_geo = "tract")

# Total workers and number of low skill and high skill workers in block
workers <- pa_wac %>% select("w_tract", "C000") %>% rename( 'GEOID' = 'w_tract')

# Get ACS census data including geometry column to map the results. Join the workers to the ACS data
phil_tract <- get_acs(
  geography = "tract",
  variables = "S0101_C01_026",
  state = "PA",
  county = c("Philadelphia"),
  year = 2020,
  geometry = TRUE
)

jobs_tracts <- phil_tract %>%
  left_join(workers)

# Convert to points
tract_pts <- sf::st_centroid(jobs_tracts)
tract_pts <- sf::st_transform(tract_pts, 4326)
phil_tract <- phil_tract %>% rename ("id" = "GEOID")
tract_pts <- tract_pts %>% rename ("id" = "GEOID")
tract_pts <- tract_pts %>% rename ('jobs' = 'C000')


# download OSM data to temp dir
mydir <- tempdir()
dest_file <- paste0(mydir,'/pennsylvania-latest.osm.pbf')

httr::GET(url = 'https://download.geofabrik.de/north-america/us/pennsylvania-latest.osm.pbf',
          httr::write_disk(dest_file),
          httr::progress())


# Build the transport network
r5r_core <- setup_r5(data_path = mydir, verbose = TRUE)

```

## Solution using the `accessibility` package
```{r}

# Calculate travel time matrix
ttm <- r5r::travel_time_matrix(
  r5r_core,
  origins = tract_pts,
  destinations = tract_pts,
  mode = "WALK",
  departure_datetime = as.POSIXct(
    "10-09-2022 9:00:00",
    format = "%d-%m-%Y %H:%M:%S"
  ),
  max_walk_time = 60,
  max_trip_duration = 60,
  verbose = TRUE,
  progress = TRUE
)


# Cumulative Opportunities
access_to_jobs <- cumulative_cutoff(
  travel_matrix = ttm,
  land_use_data = tract_pts,
  opportunity = "jobs",
  travel_cost = "travel_time_p50",
  cutoff = 60
)

# map results
access_to_jobs_tract <- phil_tract %>% left_join(access_to_jobs, by = "id")

access_map <- ggplot() +
  geom_sf(
    data = access_to_jobs_tract,
    aes(fill = jobs)
  )

access_map

```
Here's how the ouput looks like. There is a big blob of areas with missing values. Funny enough, this area gets bigger (i.e. we get more census tracts with missing values) as we consider larger `cutoff` values, say `20`, `30` or `60`.
![Rplot](https://github.com/ipeaGIT/accessibility/assets/7448421/0b9c6cdd-e4b8-4ceb-9e84-f6123594575e)

## Solution using only the `r5r` package

```{r}

# Cumulative Opportunities
accss <- r5r::accessibility(
  r5r_core,
  origins = tract_pts,
  destinations = tract_pts,
  mode = "WALK",
  departure_datetime = as.POSIXct(
    "10-09-2022 9:00:00",
    format = "%d-%m-%Y %H:%M:%S"
  ),
  max_walk_time = 60,
  max_trip_duration = 60,
  decay_function = 'step',
  opportunities_colnames = 'jobs',
  cutoffs = 60,
  verbose = TRUE,
  progress = TRUE
)

# map results
access_to_jobs_tract2 <- phil_tract %>% left_join(accss, by = "id")

# make map
access_map2 <- ggplot() +
  geom_sf(
    data = access_to_jobs_tract2,
    aes(fill = accessibility)
  )

access_map2
```
By contrast, this output has no missing values.

![Rplot01](https://github.com/ipeaGIT/accessibility/assets/7448421/7fd94a1d-e465-45ce-8697-52a28dbaaef5)

**Comment by rafapereirabr** [comment]
On a closer inspection, I've found that this occurs because the number of jobs in a few census tracts have value `NA`.  Example below. So, whenever we calculate the accessibility level from a given origin that can reach one or more locations with missing jobs, the result is `NA`. 
```
subset(tract_pts, id == 42101980500)

#> Simple feature collection with 1 feature and 6 fields
#> Geometry type: POINT
#> Dimension:     XY
#> Bounding box:  xmin: -75.14391 ymin: 40.01753 xmax: -75.14391 ymax: 40.01753
#> Geodetic CRS:  WGS 84
#>              id                                                 NAME      variable estimate moe jobs
#> 301 42101980500 Census Tract 9805, Philadelphia County, Pennsylvania S0101_C01_026        0  11   NA
#> geometry
#> 301 POINT (-75.14391 40.01753)
``` 

# Solutions
I believe there are three alternative solutions here.
1. We replace any `NA` values with `0` in the `opportunity` column. This could be done inside the `merge_by_reference()` interna function, which would cascade the desired effect to all functions in the package.
2. We could add a `na.rm = TRUE` to each of function in the package.

In both of these cases, I believe we should print a `warning` message indicating that there are missing values in the opportunity column.

3. A third alternative would be to throw and error with an informative message indicating that there are missing values in the opportunity column. This is a more radical solution. Not sure I like this one.

**Comment by dhersz** [comment]
done in https://github.com/ipeaGIT/accessibility/commit/9acc626bd99a7b69f02912759947703f7681f831

can you please test with the dev version to see the behaviour?

**Comment by rafapereirabr** [comment]
Ok, I get the warning message now, which is good. However, the output is still returning areas with `NA` values.

![Rplot02](https://github.com/ipeaGIT/accessibility/assets/7448421/f9baa328-e54b-4d5c-a9f5-ccd1e8add6d4)

**Comment by rafapereirabr** [comment]
ah, fair point. Closing issue now.
